Ex5/Graph.java

```diff
@@ -0,0 +1,99 @@
+import java.util.LinkedList;
+import java.util.List;
+
+public class Graph {

```

Comment:
Does not have to be public. It is an implementation detail of Ex5 class and is quite specialized for this purpose, so can not be easily reused by other modules. It is a good practice to hide everything unless it is usable outside of your module/package.